src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/tke_dissipation_stability_functions.jl

```diff
@@ -0,0 +1,289 @@
+
+@inline function tke_stability_functionᶜᶜᶠ(i, j, k, grid, closure::ConstantSchmidtStabilityTDVD, args...)
+    Cσe = closure.stability_functions.Cσe
+    𝕊u = momentum_stability_functionᶜᶜᶠ(i, j, k, grid, closure, args...)
+    return 𝕊u / Cσe
+end
+        
+@inline function dissipation_stability_functionᶜᶜᶠ(i, j, k, grid, closure::ConstantSchmidtStabilityTDVD, args...)
+    Cσϵ = closure.stability_functions.Cσϵ
+    𝕊u = momentum_stability_functionᶜᶜᶠ(i, j, k, grid, closure, args...)
+    return 𝕊u / Cσϵ
+end
+
+Base.@kwdef struct ConstantStabilityFunctions{FT} <: AbstractConstantSchmidtStabilityFunctions
+    Cσe :: FT = 1.0
+    Cσϵ :: FT = 1.2
+    Cu₀ :: FT = 0.53 # √3
+    Cc₀ :: FT = 0.53 # √3
+    𝕊u₀ :: FT = 0.53 # √3
+end
+
+Base.summary(s::ConstantStabilityFunctions{FT}) where FT = "ConstantStabilityFunctions{$FT}"
+
+summarize_stability_functions(s::ConstantStabilityFunctions{FT}, prefix="", sep="│   ") where FT =
+    string(prefix, "ConstantStabilityFunctions{$FT}:", '\n',
+           "    ├── 𝕊u₀: ", prettysummary(s.𝕊u₀), '\n',
+           "    ├── Cσe: ", prettysummary(s.Cσe), '\n',
+           "    ├── Cσϵ: ", prettysummary(s.Cσϵ), '\n',
+           "    ├── Cu₀: ", prettysummary(s.Cu₀), '\n',
+           "    └── Cc₀: ", prettysummary(s.Cc₀))
+
+const ConstantStabilityTDVD = TKEDissipationVerticalDiffusivity{<:Any, <:Any, <:ConstantStabilityFunctions}
+
+@inline momentum_stability_functionᶜᶜᶠ(i, j, k, grid, c::ConstantStabilityTDVD, args...) = c.stability_functions.Cu₀
+@inline   tracer_stability_functionᶜᶜᶠ(i, j, k, grid, c::ConstantStabilityTDVD, args...) = c.stability_functions.Cc₀
+
+struct VariableStabilityFunctions{FT} <: AbstractConstantSchmidtStabilityFunctions
+    Cσe :: FT
+    Cσϵ :: FT
+    Cu₀ :: FT
+    Cu₁ :: FT
+    Cu₂ :: FT
+    Cc₀ :: FT
+    Cc₁ :: FT
+    Cc₂ :: FT
+    Cd₀ :: FT
+    Cd₁ :: FT
+    Cd₂ :: FT
+    Cd₃ :: FT
+    Cd₄ :: FT
+    Cd₅ :: FT
+    𝕊u₀ :: FT
+end
+
+function VariableStabilityFunctions(FT=Float64; 
+                                    Cσe = 1.0,
+                                    Cσϵ = 1.2,
+                                    Cu₀ = 0.1067,
+                                    Cu₁ = 0.0173,
+                                    Cu₂ = -0.0001205,
+                                    Cc₀ = 0.1120,
+                                    Cc₁ = 0.003766,
+                                    Cc₂ = 0.0008871,
+                                    Cd₀ = 1.0,
+                                    Cd₁ = 0.2398,
+                                    Cd₂ = 0.02872,
+                                    Cd₃ = 0.005154,
+                                    Cd₄ = 0.006930,
+                                    Cd₅ = -0.0003372,
+                                    𝕊u₀ = nothing)
+
+    if isnothing(𝕊u₀)
+        a = Cd₅ - Cu₂
+        b = Cd₂ - Cu₀
+        c = Cd₀
+        𝕊u₀ = (2a / (-b - sqrt(b^2 - 4a * c)))^(1/4)
```

Comment:
where does this come from? reads like the solution of a quadratic polynomial somehow...

Reply:
(a comment next to the expression in the code might be helpful?)

Reply:
yes that's right, that's what it is. This is a log layer solution where production balances dissipation, I will write a comment. There's no good reference in the literature surprisingly